cekit/descriptor/packages.py

```diff
@@ -42,40 +54,57 @@ class Repository(Descriptor):
     """Object representing package repository
 
     Args:
-      name - repository name as referenced in cekit config file
+      descriptor - repository name as referenced in cekit config file
     """
 
-    def __init__(self, name):
-        self.schemas = repository_schema
+    def __init__(self, descriptor):
+        # we test parameter is not dict asi there is no easy way how to test
+        # if something is string both in py2 and py3
+        if not isinstance(descriptor, dict):
+            descriptor = self._create_repo_object(descriptor)
+
+        if 'filename' not in descriptor:
+            descriptor['filename'] = '%s.repo' % descriptor['name'].replace(' ', '_')
+        if 'url' not in descriptor:
+            descriptor['url'] = {}
+
+        self.schemas = [repository_schema]
+        super(Repository, self).__init__(descriptor)
 
+        if 'present' not in self._descriptor:
+            self._descriptor['present'] = True
+
+    def _create_repo_object(self, repository):
+        logger.warning("The way of defining repository '%s' is deprecated. Convert "
+                       "it to an URL based repository object. Consult Cekit docs, "
+                       "for more details." % repository)
```

Comment:
Link would be valuable here.

Reply:
we don't have any permalinks now :( So it will be probably added later